File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    player_dict = {}
    
    for idx, player in enumerate(players):
        player_dict[player] = idx
    
    for c in callings:
        idx = player_dict[c]
        player_dict[c] -= 1
        player_dict[players[idx - 1]] += 1
        
        players[idx-1], players[idx] = players[idx], players[idx-1]
    
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    map<string, int> m1;
    // vector<string> answer(players);
    
    for (int i = 0; i<players.size(); i++) {
        m1[players[i]] = i;
    }
    
    for(int i=0;i<callings.size();i++){
        int locate = m1[callings[i]];
        m1[players[locate-1]] +=1;
        m1[callings[i]] -=1;
        // answer에 저장된 순서 변경
        string tmp = players[locate];
        players[locate] = players[locate-1];
        players[locate-1] = tmp; 
    }
    
    return players;
}